File: Api/LeaveRequests/Models/LeaveRequest.cs
﻿using Api.Employees.Models;

namespace Api.LeaveRequests.Models;

public record LeaveRequest
{
    public int Id { get; init; }

    public required EmployeeBrief Employee { get; init; }

    public required string AbsenceReason { get; init; }

    public DateTime StartDate { get; init; }

    public DateTime EndDate { get; init; }

    public string? Comment { get; init; }

    public required string Status { get; init; }
}

File: client/src/leaveRequests/models/LeaveRequest.ts
import ApprovalRequest from "src/approvalRequests/models/ApprovalRequest";
import Employee from "src/employees/models/Employee";

type LeaveRequest = {
  id: number;
  startDate: string;
  endDate: string;
  absenceReason: string;
  status: string;
  comment?: string;
  employee: Employee;
  approvalRequests: ApprovalRequest[];
}

export default LeaveRequest;